### PR TITLE
update cm key.

### PR DESCRIPTION
**Pull request by droslean**
Update the prowgen to use a single configMap for the prow jobs, and reformat the key to `$org-$repo-$branch` (without the `.yaml`)

### CODE REVIEW

File: cmd/ci-operator-prowgen/main.go [hunk 1]
```diff
@@ -79,9 +79,9 @@ func generatePodSpec(org, repo, configFile, target string, additionalArgs ...str
 	configMapKeyRef := kubeapi.EnvVarSource{
 		ConfigMapKeyRef: &kubeapi.ConfigMapKeySelector{
 			LocalObjectReference: kubeapi.LocalObjectReference{
-				Name: fmt.Sprintf("ci-operator-%s-%s", org, repo),
+				Name: "ci-operator-configs",
 			},
-			Key: configFile,
+			Key: org + "-" + repo + "-" + configFile[0:len(configFile)-len(filepath.Ext(configFile))],
```
Comment (bbguimaraes) [inline cmd/ci-operator-prowgen/main.go]: Pass the branch name as an argument instead of re-parsing the file name here.

**Review by bbguimaraes — CHANGES_REQUESTED** [review]

File: cmd/ci-operator-prowgen/main.go (hunk 1 above)
Comment (droslean) [reply cmd/ci-operator-prowgen/main.go]: the branch name is the filename

Comment (droslean) [reply cmd/ci-operator-prowgen/main.go]: I am not parsing it, I am just removing the extension here (`.yaml`)

Comment (petr-muller) [reply cmd/ci-operator-prowgen/main.go]: Consider using `Sprintf` and `TrimSuffix` (https://golang.org/pkg/strings/#TrimSuffix) for readability

**Comment by bbguimaraes** [comment]
We need a corresponding PR in `openshift/release` ready so we can merge this without blocking other PRs.

/hold

**Comment by droslean** [comment]
after realizing that this PR is not serving prow's config updater, I am closing it.

**Pull request closed without merging**